go/vt/vttablet/tabletserver/tabletserver.go

```diff
@@ -2044,7 +2046,12 @@ func queryAsString(sql string, bindVariables map[string]*querypb.BindVariable, s
 		}
 	}
 	fmt.Fprintf(buf, "}")
-	return buf.String()
+	query := buf.String()
+	if truncateForLog {
+		return sqlparser.TruncateForLog(query)
+	} else {
+		return query
+	}
 }
```

Comment:
with this change `truncateSQLAndBindVars` and `queryAsString` looks similar. Those methods should be merged.

Reply:
Good call. I've pushed an update that merges `truncateSQLAndBindVars` and `queryAsString`. I am still working on adding unit tests though. Will let you know once I've added them.